### PR TITLE
FindConditionByType func added

### PR DESCRIPTION
**Pull request by nurali-techie**

### CODE REVIEW

File: pkg/condition/condition.go [hunk 1]
```diff
@@ -20,6 +20,15 @@ func AddOrUpdateStatusConditions(conditions []toolchainv1alpha1.Condition, newCo
 	return conditions, atLeastOneUpdated
 }
 
+func FindConditionByType(conditions []toolchainv1alpha1.Condition, conditionType toolchainv1alpha1.ConditionType) toolchainv1alpha1.Condition {
```
Comment (xcoulon) [inline pkg/condition/condition.go]: please add a test for this function ;)

Comment (nurali-techie) [reply pkg/condition/condition.go]: @xcoulon thanks, will add :+1:

Comment (alexeykazakov) [inline pkg/condition/condition.go]: It's not clear what this function returns if no condition found. You will have to look at the function implementation to figure it out.
Let's modify it to
```
func FindConditionByType(conditions []toolchainv1alpha1.Condition, conditionType toolchainv1alpha1.ConditionType) (toolchainv1alpha1.Condition, bool)
```
where the returned bool is true if found.
OR
```
func FindConditionByType(conditions []toolchainv1alpha1.Condition, conditionType toolchainv1alpha1.ConditionType) *toolchainv1alpha1.Condition
```
And also please add a description (and a test as Xavier mentioned).

Comment (nurali-techie) [reply pkg/condition/condition.go]: Done https://github.com/codeready-toolchain/toolchain-common/pull/36/commits/a247e08b1bed03a3437043ae547c370912ad3eaa

Comment (nurali-techie) [reply pkg/condition/condition.go]: Done https://github.com/codeready-toolchain/toolchain-common/pull/36/commits/a247e08b1bed03a3437043ae547c370912ad3eaa

**Comment by nurali-techie** [comment]
@xcoulon @alexeykazakov thanks for review, comments are fixed.  kindly re-review.

**Pull request merged**